config/v1/types.go

```diff
@@ -80,6 +81,120 @@ type ConfigMapReference struct {
+	// BuildDefaults controls the default information for Builds
+	// +optional
+	BuildDefaults BuildDefaults `json:"buildDefaults,omitempty"`
+	// BuildOverrides controls override settings for builds
+	// +optional
+	BuildOverrides BuildOverrides `json:"buildOverrides,omitempty"`
+}
+
+type BuildDefaults struct {
+	// GitHTTPProxy is the location of the HTTPProxy for Git source
+	// +optional
+	GitHTTPProxy string `json:"gitHTTPProxy,omitempty"`
+
+	// GitHTTPSProxy is the location of the HTTPSProxy for Git source
+	// +optional
+	GitHTTPSProxy string `json:"gitHTTPSProxy,omitempty"`
+
+	// GitNoProxy is the list of domains for which the proxy should not be used
+	// +optional
+	GitNoProxy string `json:"gitNoProxy,omitempty"`
+
+	// Env is a set of default environment variables that will be applied to the
+	// build if the specified variables do not exist on the build
+	// +optional
+	Env []corev1.EnvVar `json:"env,omitempty"`
+
+	// SourceStrategyDefaults are default values that apply to builds using the
+	// source strategy.
+	// +optional
+	SourceStrategyDefaults SourceStrategyDefaults `json:"sourceStrategyDefaults,omitempty"`
+
+	// ImageLabels is a list of docker labels that are applied to the resulting image.
+	// User can override a default label by providing a label with the same name in their
+	// Build/BuildConfig.
+	// +optional
+	ImageLabels []ImageLabel `json:"imageLabels,omitempty"`
+
+	// Annotations are annotations that will be added to the build pod
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// Resources defines resource requirements to execute the build.
+	// +optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+}
+
+type SourceStrategyDefaults struct {
+
+	// Incremental indicates if s2i build strategies should perform an incremental
+	// build or not
+	// +optional
+	Incremental bool `json:"incremental,omitempty"`
+}
+
+type ImageLabel struct {
+	// Name defines the name of the label. It must have non-zero length.
+	Name string `json:"name"`
+
+	// Value defines the literal value of the label.
+	// +optional
+	Value string `json:"value,omitempty"`
+}
+
+type BuildOverrides struct {
+	// ForcePull indicates whether the build strategy should always force pull images
+	// +optional
+	ForcePull bool `json:"forcePull,omitempty"`
+
+	// ImageLabels is a list of docker labels that are applied to the resulting image.
+	// If user provided a label in their Build/BuildConfig with the same name as one in this
+	// list, the user's label will be overwritten.
+	// +optional
+	ImageLabels []ImageLabel `json:"imageLabels,omitempty"`
+
+	// NodeSelector is a selector which must be true for the build pod to fit on a node
+	// +optional
+	NodeSelector metav1.LabelSelector `json:"nodeSelector,omitempty"`
+
+	// Annotations are annotations that will be added to the build pod
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
```

Comment:
let's drop this. it was previously need to set tolerations, before tolerations became a first class field.  I don't expect anyone is using it anymore.